apps/developer-portal/src/components/user-stores/edit/edit-basic-details-user-store.tsx

```diff
@@ -52,79 +58,419 @@ export const EditBasicDetailsUserStore = (
+        </ConfirmationModal>
+    );
+
     return (
-        <Grid>
-            <Grid.Row columns={ 1 }>
-                <Grid.Column width={ 8 }>
-                    <Forms
-                        onSubmit={ (values: Map<string, FormValue>) => {
-                            const data = { ...userStore };
-                            data.description = values.get("description").toString();
-                            data.name = values.get("name").toString();
-                            delete data.typeName;
-                            delete data.className;
-
-                            updateUserStore(id, data).then(() => {
-                                dispatch(addAlert({
-                                    description: "The Userstore has been updated successfully.",
-                                    level: AlertLevels.SUCCESS,
-                                    message: "Userstore updated successfully!"
-                                }));
-                                update();
-                            }).catch((error) => {
-                                dispatch(addAlert({
-                                    description: error?.description ?? "An error occurred while updating the Userstore",
-                                    level: AlertLevels.ERROR,
-                                    message: error?.message ?? "Something went wrong"
-                                }));
-                            });
-                        } }
-                        submitState={ submit }
-                    >
-                        <Field
-                            label="Name"
-                            name="name"
-                            type="text"
-                            required={ false }
-                            requiredErrorMessage="Name is a required field"
-                            placeholder="Enter a name"
-                            value={ userStore?.name }
-                            disabled
-                        />
-                        <Field
-                            label="Type"
-                            name="type"
-                            type="text"
-                            disabled
-                            required={ false }
-                            requiredErrorMessage="Select a Type"
-                            value={ userStore?.typeName }
-                        />
-                        <Field
-                            label="Description"
-                            name="description"
-                            type="textarea"
-                            required={ false }
-                            requiredErrorMessage=""
-                            placeholder="Enter a description"
-                            value={ userStore?.description }
+        <>
+            { confirmDelete && deleteConfirmation() }
+            <Forms
+                onSubmit={ (values: Map<string, FormValue>) => {
```

Comment:
WDYT about separating this out to a handler function?
Check other places as well.